torch/csrc/jit/passes/onnx/peephole.cpp

```diff
@@ -22,7 +22,7 @@ std::unordered_set<NodeKind> broadcasting = {
 
 bool isNopTranspose(const std::vector<int64_t> & perm) {
   for (size_t i = 0; i < perm.size(); i++)
-    if (perm[i] != (int64_t) i)
+    if (static_cast<size_t>(perm[i]) != i)
```

Comment:
nit: it would be better to make the loop counter an `int64_t` instead of casting at every step

Reply:
Then `perm.size()` should be casted to `int64_t` as well. If I put it in the condition of the for loop, the cast will be done at each iterations as well? Or should I do the cast before the loop?

Reply:
Here's a nice way to do this:
```cpp
for (int64_t i = 0, perm_size = perm.size(); i < perm_size; i++)
  // loop
```